spyder/widgets/variableexplorer/tests/test_dataframeeditor.py

```diff
@@ -174,6 +179,34 @@ def test_change_format_with_format_not_starting_with_percent(qtbot, monkeypatch)
     with qtbot.assertNotEmitted(editor.sig_option_changed):
         editor.change_format()
 
+def test_header_bom():
+    if PY2:
+        df = read_csv(os.path.join(FILES_PATH, 'issue_2514.csv'))
+    else:
+        df = read_csv(os.path.join(FILES_PATH, 'issue_2514.csv'),
+                      encoding='utf-8-sig')
```

Comment:
Given that tests are not passing without setting an encoding in PY3, I think we should remove the conditions we put in `headerData` to make things PY2 only.

@dalthviz, what do you think?